ntp-proto/src/packet/extensionfields.rs

```diff
@@ -627,7 +636,14 @@ impl<'a> Iterator for ExtensionFieldStreamer<'a> {
     }
 }
 
-const fn next_multiple_of(lhs: u16, rhs: u16) -> u16 {
+const fn next_multiple_of_u16(lhs: u16, rhs: u16) -> u16 {
+    match lhs % rhs {
+        0 => lhs,
+        r => lhs + (rhs - r),
+    }
+}
+
+const fn next_multiple_of_usize(lhs: usize, rhs: usize) -> usize {
```

Comment:
I suggest investing some time to make this generic. If need be I can take that up (I remember having done that before but can't find where anymore).